.gitignore

```diff
@@ -1,2 +1,31 @@
 .vs
 build
+
+# CMake
+CMakeLists.txt.user
+CMakeCache.txt
+
+# VSCode
+.vscode
+*.code-workspace
+# Local History for Visual Studio Code
+.history/
+
+# Linux
+# .nfs files are created when an open file is removed but is still being accessed
+.nfs*
+
+# MacOS
+._*
+.DS_Store
+.AppleDouble
+.LSOverride
+xcuserdata/
+
+# Windows
+Thumbs.db
+Thumbs.db:encryptable
+ehthumbs.db
+ehthumbs_vista.db
+# Folder config file
+[Dd]esktop.ini
```

Comment:
Why do you have all these files in the root directory, BTW? Are you not using the build folder for your build files?

Reply:
Most of these files are very common additions to git ignores because they are files created by the OS for varius reasons. I can remove the cmake files if you perfer.

For Linux the `.ntfs*` catches temporary files from network file shares.
Most of the macOS ones are caching of various folder information, with `.DS_Store` being very, very commonly created when interacting with a folder, and for Windows, the thumbnail files are mostly for older versions of Windows as far as I know, but I'm not that familiar with development on that platform

Reply:
Yes, I see. I was just trying to keep .gitignore as lean as possible, and on all platforms if you generate build files in build folder, all of them were ignored.
VSCode and Linux make sense to me.  For Windows and Mac, should these files not be in the build folder?

Reply:
`.DS_Store` for example will usually be generated if you were to example look at the repo root in Finder.
The reason is that the OS likes to cache folder attributes like icons and other stuff for easy access.
Same with windows for some versions if you open a folder with pictures in, as Explorer.exe generates thumbnails they are cached in a `Thumbs.db`. Neither has anything to do with build artifacts. `xcuserdata/` is arguably not needed, it is only likely to appear if you actually use Xcode as an IDE, most people I know on macOS only use it as a build tool, however.

If you would rather keep it lean, I can remove this stuff, it's just a minor development quality of life thing after all, does not affect using the library.